### PR TITLE
Monitoring doc generation is no longer done by the CI but with code generation

### PR DESCRIPTION
**Pull request by vpnachev**
**How to categorize this PR?**

/area dev-productivity  documentation monitoring  open-source
/kind enhancement 

**What this PR does / why we need it**:
Monitoring doc generation is no longer done by the CI but with the code generation, i.e. `make generate`.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
https://github.com/gardener/gardener/blob/d511ef2db9555f00e172a1b5e065c040886350bf/branch.cfg#L82-L95 needs to be removed after this PR is merged. 

/cc @wyb1 @istvanballok 

**Release note**:

```other operator
NONE
```

### CODE REVIEW

File: hack/generate-monitoring-docs.sh [hunk 1]
```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env bash
```
Comment (stoyanr) [inline hack/generate-monitoring-docs.sh:1]: Missing copyright header

File: hack/generate-monitoring-docs.sh [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+set -o errexit
+
+echo "> Generating monitoring docs..."
```
Comment (stoyanr) [inline hack/generate-monitoring-docs.sh]: Nit: for consistency with the rest of the output of `make generate`
```suggestion
echo "Generating monitoring docs"
```

**Review by stoyanr — CHANGES_REQUESTED** [review]
Only 2 small suggestions. Tested the code generation locally and found no issues.

File: hack/generate-monitoring-docs.sh (hunk 2 above)
Comment (vpnachev) [reply hack/generate-monitoring-docs.sh]: Based on https://github.com/gardener/gardener/blob/848c7a187d879fa97a4246a1ba5a8ba41425502f/hack/generate.sh#L19, I am using 
```bash
echo "> Generate monitoring docs"
```

**Review by stoyanr — APPROVED** [review]
/lgtm

**Pull request merged**